src/main/resources/lib/configFile/parsingCallbacks.js

```diff
@@ -22,16 +26,18 @@ const parseStringArray = function(jsonStringValue) {
     })
 }
 
+exports.firstAtsToDollar = (value) => (value || '').replace(/@@\{/, '${');
+
 
 
 // Keys below are converted to regexpattern-ready strings for detecting all .cfg keys matching 'idprovider.<idprovidername>.<key>'.
 // For example, "defaultGroups" below is converted to '^idprovider\.[a-zA-Z0-9_-]+\.defaultGroups$' which will match the keys
 // 'idprovider.oidc.defaultGroups' and 'idprovider.other.defaultGroups' in the config file, but not 'idprovider.oidc.tokenUrl'.
 const IDPROVIDER_PARSE_CALLBACKS = {
-    'defaultGroups': parseStringArray,
-    'scopes':  (value) => parseStringArray(value).join(" "),
-    'mappings.displayName': (value) => ((value || '') + '').replace(/@@/g, '$'),
-    'mappings.email': (value) => ((value || '') + '').replace(/@@/g, '$')
+    'defaultGroups': exports.parseStringArray,
+    'scopes':  (value) => exports.parseStringArray(value).join(" "),
+    'mappings.displayName': exports.atsToDollar,
+    'mappings.email': exports.atsToDollar
```

Comment:
I usually try to avoid `exports.NameHere` inside own file. 🤷 
If you want it can be changed to. 
```
const parseStringArray = function() {...}
exports.parseStringArray = parseStringArray
```
Then it can still be used inside its own file.